RomanNumbers, new-recipe

cnelmortimer opened the pull request:
Adding a new program, RomanNumbers (https://github.com/cnelmortimer/RomanNumbers), to the repository of recipes.

Begasus commented:
Could you rename the PR to reflect the name of the recipe?

cnelmortimer commented:
Done @Begasus. The name now is "RomanNumbers, new-recipe". Regards.

Begasus commented:
Builds and runs ok on x86_gcc2 :+1:

Begasus commented:
Build fails on x86_64 (Beta1) with:
```
SRC/ConversionHelper.cpp: In function 'char* FromDecimalToRoman(const char*, char**)':
SRC/ConversionHelper.cpp:4:12: error: 'strlen' was not declared in this scope
  int len = strlen(val);
            ^~~~~~
SRC/ConversionHelper.cpp:4:12: note: suggested alternative: 'len'
  int len = strlen(val);
            ^~~~~~
            len
SRC/ConversionHelper.cpp:10:24: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
     *outErrorMessage = "Each digit must be a number between 0 and 9";
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp:20:24: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
     *outErrorMessage = "Number 0 cannot be expressed in roman numbers";
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp:24:22: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   *outErrorMessage = "Your decimal number must contain between 1 and 10 digits";
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp: In function 'char* FromRomanToDecimal(const char*, char**)':
SRC/ConversionHelper.cpp:30:12: error: 'strlen' was not declared in this scope
  int len = strlen(val);
            ^~~~~~
SRC/ConversionHelper.cpp:30:12: note: suggested alternative: 'len'
  int len = strlen(val);
            ^~~~~~
            len
SRC/ConversionHelper.cpp:33:22: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   *outErrorMessage = "Your roman number must contain between 1 and 41 characters";
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp:38:24: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
     *outErrorMessage = "Use only the following chars to write roman numbers: 'M', 'D', 'C', 'L', 'X', 'V', 'I', '(' and ')'";
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp:40:24: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
     *outErrorMessage = "The roman number in unbalanced. Missing parenthesis";
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SRC/ConversionHelper.cpp:44:7: error: 'strcmp' was not declared in this scope
    if(strcmp(extraTest, val)==0){
       ^~~~~~
SRC/ConversionHelper.cpp:44:7: note: suggested alternative: 'strtol'
    if(strcmp(extraTest, val)==0){
       ^~~~~~
       strtol
SRC/ExamWindow.cpp: In member function 'void ExamWindow::Correct()':
SRC/ExamWindow.cpp:106:6: error: 'strcmp' was not declared in this scope
   if(strcmp(userInput, targetRoman)==0){
      ^~~~~~
SRC/ExamWindow.cpp:106:6: note: suggested alternative: 'strtol'
   if(strcmp(userInput, targetRoman)==0){
      ^~~~~~
      strtol
makefile:2: recipe for target 'all' failed
make: *** [all] Error 1
```

cnelmortimer commented:
Mm, it seems I swapped the x32 and x64 images for the final test... :-(  I'll take a look right now. Thank you very much @Begasus

korli commented:
I would move this to haiku-apps/ or haiku-misc/

cnelmortimer commented:
Why @korli ? "haiku-apps" because it is native? I chose "sci-math" because I have always studied Roman Numbers in Maths... Who must take the decision?

pulkomandy commented:
We follow Gentoo, and "haiku-apps" is similar in spirit to their "x11-apps". I think putting everything that uses the Haiku API there is not going to scale well as we keep getting more native apps, however.

cnelmortimer commented:
Well, I have fixed the error and re-tested in x86 and x64 (it was an including problem). Do I remove this branch and pull request to create a new one in another folder from scracth? or do you prefer me to continue on this branch? Thank you in advance!

pulkomandy commented:
Please keep the pull request and the branch for a given recipe, it's simpler to track all of it in a single place.

Begasus commented:
Just push your changes here, so we can test :)

Begasus commented on sci-mathematics/romannumbers/romannumbers-1.0.recipe:
```diff
@@ -0,0 +1,46 @@
+COPYRIGHT="2019 N.C. Cruz"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="e754940ab3e06c5588b0d0a2a4064a29198a336b"
```
Just refer to your latest commit here: 09df30cf394dbbb33a3bd926a43d666b9fceecc5

Begasus commented:
Maybe rename the recipe to romannumbers-1.0~git.recipe? (as it is a git checkout, not an official release tarball)

Begasus commented on sci-mathematics/romannumbers/romannumbers-1.0.recipe:
```diff
@@ -0,0 +1,46 @@
+DESCRIPTION="This is a bidirectional roman to decimal numbers converter.   \
+The program can convert integers from 1 to 2147483647 with the parentheses\
+notation. It has a simple GUI including a challenging exam mode that features\
+two difficulty levels.                                                                                                      \
```
Remove the spaces at the end of "levels."
```diff
-two difficulty levels.                                                                                                      \
+two difficulty levels.
```
PS not using the trailing slash \ at the end also starts a new line :)

cnelmortimer commented:
Ok, I have renamed the recipe as suggested by @Begasus. The recipe is where I put it first, but I do not mind if you prefer to move it elsewhere. I just wanted to share the program as I did with Pin8. Kind regards!

Begasus commented:
LGTM, tested on x86_gcc2 and x86_64, both build and work ok, thanks for your contribution @cnelmortimer !
I'll wait for the suggestion on the path to put it, otherwise this can be merged :+1:

cnelmortimer commented:
Ok, thank you all. It's been a big pleasure!

Begasus commented:
Given the comments maybe it would be better to move this to "haiku/apps" (and the nature of the recipes in there also)

cnelmortimer commented:
Ok @Begasus, I do not mind the change. Do I have to do it? When is that decision final? (Yes, I am not used to contribute to others’ projects, sorry 😅)

Begasus commented:
It would be best if you did it, so the credentials will stay in your name ;) I no one objects and the move is done I'll merge it in a bit. :)

cnelmortimer commented:
Hi again! I think it's done: I moved the recipe from sci-mathematics to haiku-apps, and also removed the extra spacing and "\" in a certain line of the description. Kind regards!

Begasus commented:
Merged, thanks @cnelmortimer !
